intake_stac/catalog.py

```diff
@@ -101,13 +101,78 @@ def serialize(self):
         """
         return self.yaml()
 
+    def stack_items(
+        self, items, assets, path_as_pattern=None, concat_dim='band', override_coords=None
+    ):
+        """
+        Experimental. Create an xarray.DataArray from a bunch of STAC Items
+
+        Parameters
+        ----------
+        items: list of STAC item id strings
+        assets : list of strings representing the different assets
+        (assset key or eo:bands "common_name" e.g. ['B4', B5'], ['red', 'nir'])
+        path_as_pattern : pattern string to extract coordinates from asset href
+        concat_dim : name of concatenation dimension for xarray.DataArray
+        override_coords : list of custom coordinate names
+
+        Returns
+        -------
+        CombinedAssets instance with mapping of asset names to xarray coordinates
+
+        Examples
+        -------
+        source = cat.stack_items(['S2A_36MYB_20200814_0_L2A','S2A_36MYB_20200811_0_L2A'],
+                                 ['red', 'nir'])
+        da = stack().to_dask()
+        """
+        common2band = self[items[0]]._get_band_name_mapping()
+        metadatas = {'items': {}}
+        hrefs = []
+        for item in items:
+            metadatas['items'][item] = {'STAC': self[item].metadata, 'assets': {}}
```

Comment:
the way i ended up structuring this it ended up being basically a loop around `StacItem.stack_bands()` so could probably just call that method instead of a bunch of duplicate code?